src/Processors/Formats/Impl/Parquet/ParquetLeafColReader.cpp

```diff
@@ -488,17 +455,114 @@ void ParquetLeafColReader<TColumn>::readPageV1(const parquet::DataPageV1 & page)
         case parquet::Encoding::DELTA_BINARY_PACKED:
         case parquet::Encoding::DELTA_LENGTH_BYTE_ARRAY:
         case parquet::Encoding::DELTA_BYTE_ARRAY:
-            throw Exception(ErrorCodes::PARQUET_EXCEPTION, "Unsupported encoding: {}", page.encoding());
+            throw Exception(ErrorCodes::PARQUET_EXCEPTION, "Unsupported encoding: {}", enconding_type);
 
         default:
-          throw Exception(ErrorCodes::PARQUET_EXCEPTION, "Unknown encoding type: {}", page.encoding());
+            throw Exception(ErrorCodes::PARQUET_EXCEPTION, "Unknown encoding type: {}", enconding_type);
     }
 }
 
 template <typename TColumn>
-void ParquetLeafColReader<TColumn>::readPageV2(const parquet::DataPageV2 & /*page*/)
+void ParquetLeafColReader<TColumn>::readPageV1(const parquet::DataPageV1 & page)
 {
-    throw Exception(ErrorCodes::NOT_IMPLEMENTED, "read page V2 is not implemented yet");
+    static parquet::LevelDecoder repetition_level_decoder;
```

Comment:
Not related to this PR, but wait what? One LevelDecoder is shared among all decoding threads, and multiple `repetition_level_decoder.SetData` calls can happen in parallel? Why aren't we getting TSAN errors in CI? `SetData` assigns some fields but doesn't use them, so the data race shouldn't break anything other than TSAN, but we should fix it anyway.

AFAICT, all the SetData call does for us here is take the first 4 bytes of `buffer` as an int32_t.

Can you please fix it along the way? Just replace the `repetition_level_decoder.SetData` with a `::arrow::util::SafeLoadAs<int32_t>(buffer)` call just like in the other branch 11 lines under the SetData call (with two range checks, see next comment).